llvm/lib/SYCLLowerIR/LowerWGLocalMemory.cpp

```diff
@@ -0,0 +1,135 @@
+//===-- LowerWGLocalMemory.cpp - SYCL kernel local memory allocation pass -===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// See intro comments in the header.
+//===----------------------------------------------------------------------===//
+
+#include "llvm/SYCLLowerIR/LowerWGLocalMemory.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/Pass.h"
+
+using namespace llvm;
+
+#define DEBUG_TYPE "LowerWGLocalMemory"
+
+static constexpr char SYCL_ALLOCLOCALMEM_CALL[] = "__sycl_allocateLocalMemory";
```

Comment:
I assume that you are looking for calls of particular forward-declared functions and you are comparing names in the following manner:
```c++
      if (Callee->getName() != SYCL_ALLOCLOCALMEM_CALL)
        continue;
```

Please note that if `__sycl_allocateLocalMemory` is a forward-declared function somewhere in SYCL headers, then its name will be mangled (unless it is declared as `extern C`). Please check that with @sergey-semenov.

If the name is mangled, you should either use `.contains()` or (which is better) use `.startswith()` and include `_Z[some-number]` prefix into `SYCL_ALLOCLOCALMEM_CALL` reference string

Reply:
__sycl_allocateLocalMemory is declared as extern C function. So, there is no mangling and overload.